### DISCUSSION TITLE
🕹️ Help Needed: Interfacing USB dance mat with MicroPython-compatible microcontroller(s)

### DISCUSSION THREAD

**Discussion by AltoRetrato** [micropython/micropython#13240] [Hardware & Peripherals]
Hello, MicroPython World!

I have an old USB dance mat originally used for games like DDR, Pump It Up, and StepMania on a PC. I want to interface it with a microcontroller that supports MicroPython so I can use it wirelessly via Bluetooth or Wi-Fi with other hardware, such as VR headsets. While I do have experience working with microcontrollers (Arduino, ESP32, etc.), Bluetooth, and Wi-Fi, I lack experience in interfacing USB hardware with microcontrollers. I would greatly appreciate some guidance on how to achieve this connection.

Could anyone recommend microcontroller boards suitable for interfacing with USB devices like this? Additionally, any advice or resources on how to establish this connection and read data from the dance mat using MicroPython would be immensely helpful.

I'm aware that I could easily open the dance mat, connect a microcontroller to the internal wires, and completely bypass the USB interface. However, my aim is to create a device that can also connect to other USB hardware, be easily used by other people (with minimum (dis)assembly required), and use this opportunity to learn about interfacing with USB hardware as well. 😉

Thank you in advance for your assistance! 😃

**Comment by ned-pcs** [comment]
I don't think that Micropython currently has support for USB host mode.
However, you might consider Circuit Python, which does.

**Reply by AltoRetrato** [reply]
Thanks! I found the [CircuitPython  USB Host documentation](https://docs.circuitpython.org/en/latest/shared-bindings/usb_host/index.html), and there is a list of supported boards! 👍
